docs/public-api.asciidoc

```diff
@@ -247,14 +255,42 @@ NOTE: Spans created via this method can not be retrieved by calling <<api-curren
 ==== `void end()`
 Ends the transaction and schedules it to be reported to the APM Server.
 It is illegal to call any methods on a span instance which has already ended.
-This also includes this method and {@link #createSpan()}.
+This also includes this method and <<api-transaction-start-span>>.
 Example:
 
 [source,java]
 ----
 transaction.end();
 ----
 
+[float]
+[[api-transaction-activate]]
+==== `Scope activate()`
+Makes this span the active span on the current thread until `Scope#close()` has been called.
+Scopes should only be used in try-with-resource statements in order to make sure the `Scope#close()` method is called in all
+circumstances.
+Failing to close a scope can lead to memory leaks and corrupts the parent-child relationships.
+
+This method should always be used within a try-with-resources statement:
+[source,java]
+----
+Transaction transaction = ElasticApm.startTransaction();
+// Within the try block the transaction is available
+// on the current thread via ElasticApm.currentTransaction().
+// This is also true for methods called within the try block.
+try (final Scope scope = transaction.activate()) {
+    transaction.setName("MyController#myAction");
+    transaction.setType(Transaction.TYPE_REQUEST);
+    // do your thing...
+} catch (Exception e) {
+    transaction.captureException(e);
+    throw e;
+} finally {
+    transaction.end();
```

Comment:
It will be great to have annotations do all that, instead of instructing users

Reply:
That is planned for #212